DBconnect/DBconnection.cs

```diff
@@ -78,6 +143,25 @@ public void ConnectDB()
             }
         }
 
+        public async void ConnectDBAsync()
+        {
+            if (connection != null && connection.Ping())
+            {
+                error?.Invoke("Подключение уже осуществлено [async connection]");
```

Comment:
```suggestion
                error?.Invoke("Подключение уже осуществлено [async connection]");
```
Не факт, что подключение ранее было установлено асинхронно.